pkg/component/runtime/service.go

```diff
@@ -147,23 +147,22 @@ func (s *serviceRuntime) Run(ctx context.Context, comm Communicator) (err error)
 		s.stop(ctx, comm, lastCheckin, am == actionTeardown)
 	}
 
-	processTeardown := func(am actionMode, signed *component.Signed) error {
+	processTeardown := func(am actionMode, signed *component.Signed) {
 		s.log.Debugf("start teardown for %s service", s.name())
 		// Inject new signed
 		newComp, err := injectSigned(s.comp, signed)
 		if err != nil {
 			s.log.Errorf("failed to inject signed configuration for %s service, err: %v", s.name(), err)
-			return err
```

Comment:
Why don't we return on error here? If there's a reason it should be a comment as I could see this being added back later by someone without context.

Reply:
The implementation before was incorrect it was it was assigning the error returned from processTeardown to the outer scope error returned from Run
https://github.com/elastic/elastic-agent/blob/e8f79865c5c1747d9be5bf1581bc765c89cf9d13/pkg/component/runtime/service.go#L181

The ```processTeardown``` is only called from ```onTeardown``` which doesn't return any errors.

The error from ```injectSigned``` is highly unlikely, cause it remarshalls the same config map
https://github.com/elastic/elastic-agent/pull/3203/files/b7ebf651fdec16a858feb2272f9dc6e4724f7256#diff-4f68d92e6db0175cc6917df7f966feed47cfd66753553f9a86c98ee7ed184b1fR269
that was decoded few lines above
https://github.com/elastic/elastic-agent/pull/3203/files/b7ebf651fdec16a858feb2272f9dc6e4724f7256#diff-4f68d92e6db0175cc6917df7f966feed47cfd66753553f9a86c98ee7ed184b1fR269

The intent here is to log and let the data flow through the rest of the states leading to Endpoint check-in in or the Teardown timeout, which will be both leading to the final ```onStop```.

The implementation without tamper protection immediately go to ```onStop``` that leads to the Endpoint uninstall. The addition of the tamper protection requires an additional step (roundtrip) with Endpoint to let it to unprotect itself before it can be stopped and uninstalled.